scene/resources/packed_scene.cpp

```diff
@@ -2112,6 +2112,24 @@ Ref<SceneState> PackedScene::get_state() const {
 	return state;
 }
 
+bool PackedScene::prefer_compat_format() const {
+	if (state.is_null()) {
+		return true;
+	}
+
+	StringName tile_map_name = "TileMap";
+	StringName tile_map_layer_name = "TileMapLayer";
+
+	int count = state->get_node_count();
+	for (int i = 0; i < count; i++) {
+		const StringName &type = state->get_node_type(i);
+		if (type == tile_map_name || type == tile_map_layer_name) {
+			return false;
+		}
+	}
```

Comment:
That's basically a hard-coded hack. Makes me think that this should be wrapped in `DISABLE_DEPRECATED` at least.

Reply:
Hardcoding can be avoided by adding something like `prefer_compat_format()` to `Node`s. But currently only TileMapLayer needs to carry a large PackedByteArray regularly, so I think hardcoding is better than an invasive solution affecting the node API.